src/main/scala/com/campudus/tableaux/database/model/structure/TableModel.scala

```diff
@@ -40,25 +48,42 @@ class TableModel(val connection: DatabaseConnection) extends DatabaseQuery {
     } yield t
   }
 
+  private def retrieveGlobalLangtags(): Future[Seq[String]] = {
+    // TODO don't really like dependend models
+    systemModel.retrieveSetting(SystemController.SETTING_LANGTAGS)
+      .map(f => Option(f).map(f => Json.fromArrayString(f).asScala.map(_.toString).toSeq).getOrElse(Seq.empty))
```

Comment:
This looks strange. Shouldn't the retrieved setting be an `Option` itself?
You read the String into a `JsonArray`, what does `asScala` do? Make it into a `Seq[_]`? Then you map the sequence to a `Seq[String]` and have another `toSeq` ?